portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,27 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     
+    private ArrayList<String> messages;
```

Comment:
Does toJson specifically require an ArrayList or can we change the type of `messages` to List<String>?

The idea here is that you should use the most general type possible as that will give you flexibility later on if you want to change your code. For example if this were `private List<String> messages;` then you could change its type to a LinkedList just by changing the assignment on the first line of init.